saleor_gql_loader/data_loader.py

```diff
@@ -65,6 +65,117 @@ def __init__(self, auth_token, endpoint_url="http://localhost:8000/graphql/"):
         self.headers = {"Authorization": "Bearer {}".format(auth_token)}
         self.endpoint_url = endpoint_url
 
+    def update_shop_settings(self, **kwargs):
+        """update shop settings.
+
+        Parameters
+        ----------
+        **kwargs : dict, optional
+            overrides the default value set to update the shop settings refer to the
+            ShopSettingsInput graphQL type to know what can be overriden.
+
+        Raises
+        ------
+        Exception
+            when shopErrors is not an empty list
+        """
+
+        variables = {
+            "shopSettingsInput": kwargs
+        }
+
+        query = """
+            mutation ShopSettingsUpdate($shopSettingsInput: ShopSettingsInput!) {
+              shopSettingsUpdate(input: $shopSettingsInput) {
+                shopErrors {
+                        field
+                        message
+                        code
+                    }
+              }
+            }
+        """
+
+        response = graphql_request(
+            query, variables, self.headers, self.endpoint_url)
+
+        errors = response["data"]["shopSettingsUpdate"]["shopErrors"]
+        handle_errors(errors)
+
+    def update_shop_domain(self, **kwargs):
+        """update shop domain.
+
+        Parameters
+        ----------
+        **kwargs : dict, optional
+            overrides the default value set to update the shop domain refer to the
+            SiteDomainInput graphQL type to know what can be overriden.
+
+        Raises
+        ------
+        Exception
+            when shopErrors is not an empty list
+        """
+
+        variables = {
+            "siteDomainInput": kwargs
+        }
+
+        query = """
+            mutation ShopDomainUpdate($siteDomainInput: SiteDomainInput!) {
+              shopDomainUpdate(input: $siteDomainInput) {
+                shopErrors {
+                        field
+                        message
+                        code
+                    }
+              }
+            }
+        """
+
+        response = graphql_request(
+            query, variables, self.headers, self.endpoint_url)
+
+        errors = response["data"]["shopDomainUpdate"]["shopErrors"]
+        handle_errors(errors)
+
```

Comment:
simarly here we should also return an ID

Reply:
About the ShopDomain it's a bit ambiguous

The queried Shop object has a Domain type with the following return values:
```
domain {
  host
  sslEnabled
  url
}
```

but the SiteDomainInput is the following:
```
input SiteDomainInput {
  domain: String
  name: String
}
```

Reply:
as long as it returns something related to the updated / created object I think it's fine